ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,353 @@
+            bins=self.reco_energy_bins,
+        )
+
+        background_hist = estimate_background(
+            self.background_events[self.background_events["selected_gh"]],
+            reco_energy_bins=self.reco_energy_bins,
+            theta_cuts=self.theta_cuts_opt,
+            alpha=self.alpha,
+            fov_offset_min=self.bins.fov_offset_min,
+            fov_offset_max=self.bins.fov_offset_max,
+        )
+        self.sensitivity = calculate_sensitivity(
+            signal_hist, background_hist, alpha=self.alpha
+        )
+
+        # scale relative sensitivity by Crab flux to get the flux sensitivity
+        self.sensitivity["flux_sensitivity"] = self.sensitivity[
+            "relative_sensitivity"
+        ] * self.spectrum(self.sensitivity["reco_energy_center"])
+
+    def finish(self):
+        masks = {
+            "": self.signal_events["selected"],
+            "_NO_CUTS": slice(None),
+            "_ONLY_GH": self.signal_events["selected_gh"],
+            "_ONLY_THETA": self.signal_events["selected_theta"],
+        }
+        hdus = [
+            fits.PrimaryHDU(),
+            fits.BinTableHDU(self.sensitivity, name="SENSITIVITY"),
+            fits.BinTableHDU(self.theta_cuts_opt, name="THETA_CUTS_OPT"),
+            fits.BinTableHDU(self.gh_cuts, name="GH_CUTS"),
+        ]
+
+        self.log.debug("True Energy bins: %s" % str(self.true_energy_bins.value))
+        self.log.debug("FoV offset bins: %s" % str(self.fov_offset_bins.value))
+        for label, mask in masks.items():
+            effective_area = effective_area_per_energy_and_fov(
+                self.signal_events[mask],
+                self.sim_info,
+                true_energy_bins=self.true_energy_bins,
+                fov_offset_bins=self.fov_offset_bins,
+            )
+            hdus.append(
+                create_aeff2d_hdu(
+                    effective_area[..., np.newaxis],  # +1 dimension for FOV offset
+                    self.true_energy_bins,
+                    self.fov_offset_bins,
+                    extname="EFFECTIVE AREA" + label,
+                )
+            )
+            edisp = energy_dispersion(
+                self.signal_events[mask],
+                true_energy_bins=self.true_energy_bins,
+                fov_offset_bins=self.fov_offset_bins,
+                migration_bins=self.energy_migration_bins,
+            )
+            hdus.append(
+                create_energy_dispersion_hdu(
+                    edisp,
+                    true_energy_bins=self.true_energy_bins,
+                    migration_bins=self.energy_migration_bins,
+                    fov_offset_bins=self.fov_offset_bins,
+                    extname="ENERGY_DISPERSION" + label,
+                )
+            )
+        # Here we use reconstructed energy instead of true energy for the sake of
+        # current pipelines comparisons
+        bias_resolution = energy_bias_resolution(
```

Comment:
bias_resolution are benchmarks that can be computed from events (like here) or from the IRFs (functions in pyirf or methods of the irf objects in gammapy), I'd not compute them here in the IRF tool.